toolchain/diagnostics/diagnostic_kind.def

```diff
@@ -543,6 +543,13 @@ CARBON_DIAGNOSTIC_KIND(GenericMissingExplicitParameters)
 // Pattern matching diagnostics.
 CARBON_DIAGNOSTIC_KIND(TuplePatternSizeDoesntMatchLiteral)
 
+// Dataflow diagnostics.
```

Comment:
I'd expect we want a lot of different sorts of dataflow diagnostics, and I'd lean towards each significant cluster belonging on their own, rather than under a general header such as this. Perhaps we should look at these as `unused` diagnostics instead of dataflow diagnostics in general?